ArduPlane/tailsitter.cpp

```diff
@@ -343,67 +353,62 @@ bool QuadPlane::in_tailsitter_vtol_transition(void) const
+    // sudden loss of attitude control if too low
+    const float min_scale = tailsitter.gain_scaling_min;
+    float tthr = 1.25f * hover_throttle;
+
+    // reduce control surface throws at large tilt
+    // angles (assuming high airspeed)
+    // ramp down from 1 to max_atten at tilt angles over trans_angle
+    // (angles here are represented by their cosines)
+
+    // Note that the cosf call will be necessary if trans_angle becomes a parameter
+    // but the C language spec does not guarantee that trig functions can be used
+    // in constant expressions, even though gcc currently allows it.
+    constexpr float c_trans_angle = 0.9238795; // cosf(.125f * M_PI)
 
-        // limit positive and negative slew rates of applied speed scaling
-        constexpr float posTC = 2.0f;   // seconds
-        constexpr float negTC = 1.0f;   // seconds
-        const float posdelta = plane.G_Dt / posTC;
-        const float negdelta = plane.G_Dt / negTC;
-        spd_scaler = constrain_float(spd_scaler, last_spd_scaler - negdelta, last_spd_scaler + posdelta);
-        last_spd_scaler = spd_scaler;
+    // alpha = (1 - max_atten) / (c_trans_angle - cosf(radians(90)));
+    const float alpha = (1 - min_scale) / c_trans_angle;
+    const float beta = 1 - alpha * c_trans_angle;
+
+    const float c_tilt = ahrs_view->get_rotation_body_to_ned().c.z;
+    if (c_tilt < c_trans_angle) {
+        temp_spd_scaler = constrain_float(beta + alpha * c_tilt, min_scale, 1.0f);
+        // reduce throttle attenuation threshold too
+        tthr = 0.5f * hover_throttle;
+    }
+    // if throttle is above hover thrust, apply additional attenuation
+    if (throttle > tthr) {
+        const float throttle_atten = 1 - (throttle - tthr) / (1 - tthr);
+        temp_spd_scaler *= throttle_atten;
+        temp_spd_scaler = constrain_float(temp_spd_scaler, min_scale, 1.0f);
     }
 
+    // limit positive and negative slew rates of applied speed scaling
+    constexpr float posTC = 2.0f;   // seconds
+    constexpr float negTC = 1.0f;   // seconds
+    const float posdelta = plane.G_Dt / posTC;
+    const float negdelta = plane.G_Dt / negTC;
+    temp_spd_scaler = constrain_float(temp_spd_scaler, last_spd_scaler - negdelta, last_spd_scaler + posdelta);
+
     // if gain attenuation isn't active and boost is enabled
-    if ((spd_scaler >= 1.0f) && (tailsitter.gain_scaling_mask & TAILSITTER_GSCL_BOOST)) {
-        // boost gains at low throttle
-        if (is_zero(throttle)) {
-            spd_scaler = tailsitter.throttle_scale_max;
-        } else {
-            spd_scaler = constrain_float(hover_throttle / throttle, 1.0f, tailsitter.throttle_scale_max);
-        }
+    if ((temp_spd_scaler >= 1.0f) && (tailsitter.gain_scaling_mask & TAILSITTER_GSCL_BOOST) != 0) {
+        spd_scaler = MAX(1.0f, spd_scaler);
+    } else {
+        spd_scaler = temp_spd_scaler;
     }
 
-    return spd_scaler;
+    last_spd_scaler = spd_scaler;
```

Comment:
this should now be the only change of behavior from your orignal PR for your methods, I have put this here so the boost  throttle method is included in the filter, I think this makes sence?

Reply:
It looks like scaling boost is applied regardless of whether TAILSITTER_GSCL_BOOST is set
If that check is done up above, then the test here can probably just go away, I think...

Reply:
This was it preserves the behavior of your original PR (I think?) I'm happy with this bit however you like, it should only affect the TAILSITTER_GSCL_ATT_THR method.